File: Kap13/PHP/Listing39_47/MyLogger.php
<?php

namespace mylogger;

interface MyLogger {
    public function doLog($data);
}

class MyJsonLogger implements MyLogger {
    public function doLog($data) {
        echo "\n{\"message\":\"" .$data ."\"}";
    }
}

class MyPlainLogger implements MyLogger {
    public function doLog($data) {
        echo "\n" .$data;
    }
}

?>

File: Kap13/Python/Listing39_47/MyLogger.py
import abc

class MyLogger (metaclass=abc.ABCMeta):

    @abc.abstractmethod
    def __init__(self):
        return

    @abc.abstractmethod
    def doLog(self, data):
        return


class MyPlainLogger(MyLogger):
    def __init__(self):
        super().__init__()

    def doLog(self, data):
        print(data)
    

class MyJsonLogger(MyLogger):
    def __init__(self):
        super().__init__()

    def doLog(self, data):
        print("{\"message\":\"" + data + "\"}")